### ISSUE TITLE
[Android] Implement the corresponding part in Generator for optional qr code.

### ISSUE THREAD

**Issue by yunhanw-google** [project-chip/connectedhomeip#31323]
> This PR only implement the Optional field in Parser, do we have plan to implement the corresponding part in Generator in a separate PR?

That would be my preference since we can generate the qr code using other tools. The optional field parser is more critical for unblocking our companies qr code decision.

_Originally posted by @stevePalmerin in https://github.com/project-chip/connectedhomeip/issues/31251#issuecomment-1879125205_
